doc/source/reference/routines.polynomials.rst

```diff
@@ -9,24 +9,153 @@ of the `numpy.polynomial` package, introduced in NumPy 1.4.
 
 Prior to NumPy 1.4, `numpy.poly1d` was the class of choice and it is still
 available in order to maintain backward compatibility.
-However, the newer Polynomial package is more complete than `numpy.poly1d`
-and its convenience classes are better behaved in the numpy environment.
+However, the newer `polynomial package <numpy.polynomial>` is more complete
+and its `convenience classes <routines.polynomials.classes>` provide a
+more consistent, better-behaved interface for working with polynomial
+expressions.
 Therefore :mod:`numpy.polynomial` is recommended for new coding.
 
-Transition notice
------------------
-The  various routines in the Polynomial package all deal with
-series whose coefficients go from degree zero upward,
-which is the *reverse order* of the Poly1d convention.
-The easy way to remember this is that indexes
-correspond to degree, i.e., coef[i] is the coefficient of the term of
-degree i.
+Transitioning from `numpy.poly1d` to `numpy.polynomial`
+-------------------------------------------------------
 
+As noted above, the :class:`poly1d class <numpy.poly1d>` and associated
+functions defined in ``numpy.lib.polynomial``, such as `numpy.polyfit`
+and `numpy.poly`, are considered legacy and should **not** be used in new
+code.
+Since NumPy version 1.4, the `numpy.polynomial` package is preferred for
+working with polynomials.
+
+Quick Reference
+~~~~~~~~~~~~~~~
+
+The following table highlights some of the main differences between the
+legacy polynomial module and the polynomial package for common tasks.
```

Comment:
nit: I wonder if we should settle on using a capitalized "Polynomial package" or a "`np.polynomial` package" standard so it is different from a polynomial object and from the old polynomial module in this document.

Reply:
TL;DR - you're right. The terminology is very confusing. I would vote for all instances of "polynomial package" being replaced with "`np.polynomial` package".

---

That's a good point and is actually pretty tricky. One of the biggest problems in disentangling all of the various polynomial components is how overloaded the terminology is. I had been settling on "polynomial package" and "polynomial module" to distinguish between the new and old polynomial APIs, respectively as the polynomial package is a Python package (`numpy.polynomial`), whereas the old-style polynomial API (e.g. `np.poly1d`) is technically a Python module (`numpy.lib.polynomial`). This isn't very user/reader-friendly however, as it relies on the reader having a firm grasp on the distinction between Python modules and packages.

I think we also definitely be careful about the capitalization convention because "np.polynomial.Polynomial" is one of the new-style convenience classes (representing a finite power series), so referring to the "Polynomial package" might be yet *another* way for readers to be confused by the terminology.

Reply:
That makes sense, and I did get that by reading your text. But I suspect that for newcomers this might be a bit confusing. A local solution might be to add something in the beginning of the doc to the effect of
"In what follows, we will refer to the old polynomial module `numpy.lib.polynomial` as `polynomial module`, not to be confused with the new `numpy.polynomial` package." But feel free to ignore this suggestion if you think it's not necessary.

Reply:
Thanks @melissawm , I really like your suggestion! I've added a `.. note` re: terminology in e4e5a5c